politeiad/backend/gitbe/gitbe.go

```diff
@@ -1157,6 +1155,25 @@ func (g *gitBackEnd) newRecord(token []byte, metadata []backend.MetadataStream,
 	return rm, nil
 }
 
+func (g *gitBackEnd) existingTokenPrefixes() ([]string, error) {
+	vetted, unvetted, err := g.Inventory(0, 0, false, false)
```

Comment:
Pulling the full inventory every time a new proposal is submitted is horribly inefficient.  What's a better way to do this?

Reply:
Yeah I've changed it to just look at the folders in the vetted folder, and check the branches in the unvetted one.

Reply:
Would just adding a memory cache onto the gitBackEnd context that stores the token prefixes be better?  Then its just a quick cache lookup to see if there are any collisions.

Reply:
Yeah that makes sense.